### ISSUE TITLE
Xresources support?

### ISSUE THREAD

**Issue by kgilmer** [brndnmtthws/conky#880]
**Issue**

Feature request: Read color values from Xresources

**Information**

Hello, I use Conky in my desktop environment.  I consolidate theme information such as colors into Xresources, and would like Conky to read it's color values from there as well.  I do not find anything from searching that others have done this, or is supported.  So, I'm wondering if this a feature conky maintainers would be interested in implementing, or would be open to a PR?

**Comment by lasers** [comment]
We're definitely open to the PRs. :-)

**Comment by kgilmer** [comment]
Ok, I can take a crack at.  But, are you open to taking a new dependency for this to work?  Specifically:

lib: `xcb` 
headers required: `xcb/xcb.h xcb/xcb_xrm.h`

**Comment by lasers** [comment]
It's not my place to say, but from what I glean, I don't think we're against dependencies... And if turns out to be an issue, we can add build flags later to include/exclude xcb dependencies. Cheers.

**Comment by kgilmer** [comment]
I spent a bit of time looking into this.  First, the project maintainers have done an excellent job of keeping the build clean and simple.  I was able, compile, run, and get a interactive debug session from source in about 20 minutes.  For a project with such a long legacy this is not an easy thing to achieve.  Well done!

Regarding the change itself, it seems that Conky relies on Lua to do the heavy lifting of parsing the configuration file, and uses a stack-based API to traverse a tree of values held by a Lua instance during startup to retrieve the configuration values.  In reading the code I did not find a natural place where Xresource substitution logic would be cleanly added.  I do not have any experience w/ Lua but my gut says that this logic may be more cleanly added in Lua as the configuration is parsed, rather than in C when the values are read from Lua's model.  

The multi-OS support in Conky concerns me in regards to libxcb and Xresources support.  I'm not sure how these other OS platforms support Xresources, nor am I sure how to ifdef this logic only into platforms that support Xresources.  

Given that I find no natural/easy integration point for Xresources, I will not continue on with this effort at this time, but may pick it up again in the future if my Lua capacities increase.  If anyone has suggestions or ideas for a natural way to add Xresources capability to Conky, let me know!
